### ISSUE TITLE
Expiration Date exception when trying to load pass on device or simulator

### ISSUE THREAD

**Issue by tenneyb** [daxko/PassSharp#6]
Unable to parse expirationDate /Date(1487415600000-0000)/ as a date. We expect dates in "W3C date time stamp format", either "Complete date plus hours and minutes" or "Complete date plus hours, minutes and seconds". For example, 1980-05-07T10:30-05:00.

using C# DateTime on the Pass.expirationDate field, if I leave this field out (which I really can't) the pass works fine
Help :) ?!

**Comment by scurker** [comment]
@tenneyb Good catch! This is definitely an issue and something we missed during early development. We need to run a few tests around changes, but expect a fix within the next week.

**Comment by anthonycamut** [comment]
You can correct this by replace the type of the date by DateTimeOffset in Pass.cs

`public DateTimeOffset? expirationDate { get; set; }`
`public DateTimeOffset? relevantDate { get; set; }`

And to modify a little bit the ToJson function in PassWriter.cs:

`  if (name.Equals("fields"))`
`        {`
`          var fields = ((Dictionary<FieldType, List<Field>>)value)`
`            .ToDictionary(x => SerializeFieldType(x.Key), x => x.Value);`
`          jsonDict.Add(pass.type, fields);`
`        }`
`        else if (isIgnoredField(name) || isNull(value) || isAsset(value) || isEmptyList(value))`
`        {`
`          // don't include value`
`        }`
`        else if (name.Equals("relevantDate"))`
`        {`
`          // Added by Anthony to correct the format`
`          jsonDict.Add(name, pass.relevantDate.Value.ToString("yyyy-MM-ddTHH:mm:ssK")); `
`        }`
`        else if (name.Equals("expirationDate"))`
`        {`
`          // Added by Anthony to correct the format`
`          jsonDict.Add(name, pass.expirationDate.Value.ToString("yyyy-MM-ddTHH:mm:ssK"));`
`        }`
`        else`
`        {`
`          jsonDict.Add(name, value);`
`        }`
I have tested it works for me!
